airflow/api_fastapi/core_api/routes/public/task_instances.py

```diff
@@ -482,3 +484,78 @@ def get_mapped_task_instance_try_details(
         map_index=map_index,
         session=session,
     )
+
+
+@task_instances_router.patch(
+    "/{task_id}",
+    responses=create_openapi_http_exception_doc([status.HTTP_404_NOT_FOUND, status.HTTP_400_BAD_REQUEST]),
+)
+@task_instances_router.patch(
+    "/{task_id}/{map_index}",
+    responses=create_openapi_http_exception_doc([status.HTTP_404_NOT_FOUND, status.HTTP_400_BAD_REQUEST]),
+)
+def patch_task_instance(
+    dag_id: str,
+    dag_run_id: str,
+    task_id: str,
+    request: Request,
+    body: PatchTaskInstanceBody,
+    session: Annotated[Session, Depends(get_session)],
+    map_index: int = -1,
```

Comment:
We are missing the update mask.

You can take a look at others (not all of them are reworked yet), but `patch_dag_run` might be close to what you want. 

Also, if `mask` is provided, `None` values should not be ignored but set to `None`.

Reply:
Done